src/frontend/src/optimizer/plan_node/batch_iceberg_scan.rs

```diff
@@ -29,10 +31,13 @@ use crate::error::Result;
 use crate::optimizer::plan_node::expr_visitable::ExprVisitable;
 use crate::optimizer::property::{Distribution, Order};
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Educe, Debug, Clone, PartialEq)]
+#[educe(Eq, Hash)]
 pub struct BatchIcebergScan {
     pub base: PlanBase<Batch>,
     pub core: generic::Source,
+    #[educe(Hash(ignore))]
+    pub predicate: IcebergPredicate,
```

Comment:
Ignoring `IcebergPredicate` directly might be incorrect, considering that if we compare two iceberg scans with different predicates, it might return true. Actually, we should return false, right? How about override the Eq trait to return false if both predicates are not `AlwaysTrue`

Reply:
The ignore is just for `Hash`. For `Eq`, we just use the `PartialEq` implementation of `IcebergPredicate`.
I suppose you mean to override the `PartialEq` implementation of `IcebergPredicate` in this context. We can do that.

Reply:
Actually, it might be better to just panic, if both of the `IcebergPredicates` are not `AlwaysTrue`.
Because in such a scenario it could affect the correctness of `Hash`.

Reply:
We should not be checking for the equality of `BatchIcebergScan`, after the iceberg predicate pushdown pass.